manifests/init.pp

```diff
@@ -0,0 +1,69 @@
+# Class: lastpass
+#
+# This class installs the LastPass CLI and provides functions to interact with it.
+
+class lastpass (
+  $package  = $lastpass::params::package,
+  $home     = '/root/.lpass',
+  $agent_timeout = 3600,
+  $username = undef,
+  $password = undef,
+) inherits lastpass::params {
+
+  package { $package:
+    ensure => present,
+  }
+
+  file { '/usr/local/bin/lpasspw':
+    ensure => file,
+    owner  => 'root',
+    group  => 'root',
+    mode   => '0755',
+    source => "puppet:///modules/${module_name}/lpasspw",
+  }
+
+  file { $home:
+    ensure => directory,
+    mode   => '0600',
+  }
+  file { "${home}/pw":
+    ensure  => file,
+    mode    => '0400',
+    content => inline_template('<%= @password %>'),
+  }
+
+  profiled::script { 'lpass.sh':
```

Comment:
How about setting a new param for $home to $HOME/.lpass?
$HOME is usually set correctly on *nix. This would work for every user.
Then change the default to undef if/when lastpass cli is fixed or figured out, so it works without setting it.

I think the code below would allow most scenarios.

- [x] Password or not
- [x] root user or not
- [x] Every login user uses their own home

```puppet
class lastpass (
  $package       = $lastpass::params::package,
  $home          = "${::root_home}/.lpass", 
  $lpass_home    = "\$HOME/.lpass",
  $lpass_agent_timeout = 3600,
  $username      = undef,
  $password      = undef,
) inherits lastpass::params {
...

  if $home {
    file { $home:
      ensure => directory,
      mode   => '0600',
    }

    if $password {
      file { "${home}/pw":
        ensure  => file,
        mode    => '0400',
        content => $password,
      }
    }
  }
}
```

```erb
# profiled template
<%= %Q(export LPASS_HOME=#{@lpass_home}\n) if @lpass_home -%>
<%= %Q(export LPASS_AGENT_TIMEOUT=#{@lpass_agent_timeout}\n) if @lpass_agent_timeout -%>
if [ -f "${LPASS_HOME}/pw" ]; then
  export LPASS_ASKPASS=/usr/local/bin/lpasspw
fi
```